flux-desugar/src/lib.rs

```diff
@@ -153,8 +154,12 @@ pub fn desugar_fn_sig(
         // See crate level comment
         genv.map().insert_generics(def_id, generics);
 
+        // DESUGAR FN_SIG (needs to happen AFTER desugaring generics)
         let (generic_preds, fn_sig) = cx.desugar_fn_sig(&fn_sig, &mut Binders::new())?;
 
+        let ref_params = fhir::RefParams { params: fn_sig.params.clone() };
+        genv.map().insert_refparams(def_id, ref_params);
```

Comment:
Yes, this makes sense -- I wonder what your strategy is for the `rty` level though, because there are no explicit `.params` there, is your point that we should just add that as a field to the `rty::FnSig`?

Reply:
For `rty` I would add a field to `rty::Generics` and put it there as described here https://github.com/flux-rs/flux/pull/503#discussion_r1319162997